### PR TITLE
Add Swift 3.0 support.

### LINKED ISSUE #53: Swift 3 Support

### PR DESCRIPTION
**Pull request by nlutsenko**
- Updating all source files to Swift 3.0 syntax
- Updating SwiftCatalog to Swift 3.0 and iOS 10

Most of the changes are simply due to language syntax change, but there are few API changes as well to make the SDK more Swifty :)

Kudos and thanks to:
- @gerbiljames for the initial request in #55 that gets superseded now.
- @milseman for clarifying `@escaping` behavior with generic types (it's implicitly `@escaping`)

Closes #53

### CODE REVIEW

File: Sources/Core/AppEvents/AppEvent.Builtin.swift [hunk 1]
```diff
@@ -49,12 +49,12 @@ extension AppEvent {
 
    - returns: An app event that can be logged via `AppEventsLogger`.
    */
-  public static func CompletedTutorial(successful successful: BooleanType? = nil,
-                                                  valueToSum: Double? = nil,
-                                                  extraParameters: ParametersDictionary = [:]) -> AppEvent {
+  public static func completedTutorial(successful: DarwinBoolean? = nil,
```
Comment (richardjrossiii) [inline Sources/Core/AppEvents/AppEvent.Builtin.swift]: `DarwinBoolean` probably isn't the right choice here.

File: Sources/Core/AppEvents/AppEvent.Builtin.swift [hunk 2]
```diff
@@ -91,16 +91,16 @@ extension AppEvent {
...
    - returns: An app event that can be logged via `AppEventsLogger`.
    */
-  public static func Searched(contentId contentId: String? = nil,
-                                        searchedString: String? = nil,
-                                        successful: BooleanType? = nil,
-                                        valueToSum: Double? = nil,
-                                        extraParameters: ParametersDictionary = [:]) -> AppEvent {
+  public static func searched(contentId: String? = nil,
+                              searchedString: String? = nil,
+                              successful: DarwinBoolean? = nil,
```
Comment (richardjrossiii) [inline Sources/Core/AppEvents/AppEvent.Builtin.swift]: `DarwinBoolean`

File: Sources/Core/AppEvents/AppEvent.Builtin.swift [hunk 3]
```diff
@@ -206,12 +205,12 @@ extension AppEvent {
 
    - returns: An app event that can be logged via `AppEventsLogger`.
    */
-  public static func AddedPaymentInfo(successful successful: BooleanType? = nil,
-                                                 valueToSum: Double? = nil,
-                                                 extraParameters: ParametersDictionary = [:]) -> AppEvent {
+  public static func addedPaymentInfo(successful: DarwinBoolean? = nil,
```
Comment (richardjrossiii) [inline Sources/Core/AppEvents/AppEvent.Builtin.swift]: `DarwinBoolean`

File: Sources/Core/GraphRequest/GraphRequestProtocol.swift [hunk 4]
```diff
@@ -59,7 +59,7 @@ extension GraphRequestProtocol {
 
    - parameter completion: Optional completion closure that is going to be called when the connection finishes or fails.
    */
-  public func start(completion: ((httpResponse: NSHTTPURLResponse?, result: GraphRequestResult<Self>) -> Void)? = nil) {
+  public func start(_ completion: @escaping GraphRequestConnection.Completion<Self>) {
```
Comment (richardjrossiii) [inline Sources/Core/GraphRequest/GraphRequestProtocol.swift]: Shouldn't this be `GraphRequestConnection<Self>.Completion`? The typealias itself doesn't have the generic paramter, `GraphRequestConnection` does, no?

File: Sources/Core/UserProfile/UserProfile.PictureView.swift [hunk 5]
```diff
@@ -24,12 +24,12 @@ import FBSDKCoreKit.FBSDKProfilePictureView
...
   /// A view to display a profile picture.
-  public class PictureView: UIView {
+  public final class PictureView: UIView {
 
-    private let sdkProfilePictureView = FBSDKProfilePictureView(frame: .zero)
+    fileprivate let sdkProfilePictureView = FBSDKProfilePictureView(frame: .zero)
 
     /// The aspect ratio of the source image of the profile picture.
-    public var pictureAspectRatio = UserProfile.PictureAspectRatio.Square {
+    open var pictureAspectRatio = UserProfile.PictureAspectRatio.square {
```
Comment (richardjrossiii) [inline Sources/Core/UserProfile/UserProfile.PictureView.swift]: `open` here should do nothing, as `PictureView` is final, no?

File: Sources/Login/LoginManager.swift [hunk 6]
```diff
@@ -98,31 +98,31 @@ public class LoginManager {
...
+                     from: viewController,
+                     handler: LoginManager.sdkCompletionFor(completion))
   }
 
   /**
    Logs the user out.
    This calls `AccessToken.current = nil` and `Profile.current = nil`.
    */
-  public func logOut() {
+  open func logOut() {
```
Comment (richardjrossiii) [inline Sources/Login/LoginManager.swift]: Again, `open` on a final class?

File: Sources/Core/AppEvents/AppEvent.Builtin.swift (hunk 1 above)
Comment (nlutsenko) [inline Sources/Core/AppEvents/AppEvent.Builtin.swift]: Good catch, need to update.

File: Sources/Core/AppEvents/AppEvent.Builtin.swift (hunk 2 above)
Comment (nlutsenko) [inline Sources/Core/AppEvents/AppEvent.Builtin.swift]: Good catch, need to update.

File: Sources/Core/AppEvents/AppEvent.Builtin.swift (hunk 3 above)
Comment (nlutsenko) [inline Sources/Core/AppEvents/AppEvent.Builtin.swift]: Good catch, need to update.

File: Sources/Core/GraphRequest/GraphRequestProtocol.swift (hunk 4 above)
Comment (nlutsenko) [inline Sources/Core/GraphRequest/GraphRequestProtocol.swift]: Yes, will update.

File: Sources/Core/UserProfile/UserProfile.PictureView.swift (hunk 5 above)
Comment (nlutsenko) [inline Sources/Core/UserProfile/UserProfile.PictureView.swift]: I knew I missed one of these.

File: Sources/Login/LoginManager.swift (hunk 6 above)
Comment (nlutsenko) [inline Sources/Login/LoginManager.swift]: Yup, should be `public`.

**Comment by nlutsenko** [comment]
@richardjrossiii, updated and rebased:
- Fixed all violations from `open` to `public`
- Replaced `DarwinBoolean` with `Bool`
- Made `completion` optional again on `GraphRequestProtocol.start()`.

**Comment by aliabadi** [comment]
When I add the Facebook dependencies to my Podfile it loads an older version of the SDK.

Is there something I'm missing?  The readme says that's all you should have to do.  I've tried copying all the newer swift files in, but that was fraught with errors...I was able to get it working by using the legacy swift option.

Thanks!

**Comment by ricardochavarria** [comment]
This seems to be merged into *master*, but I can't get it to compile with Swift 3.0 in Xcode 8.0.

**Comment by simonstewart** [comment]
@ricardochavarria I'm seeing this issue as of today.  Have you worked around it or still waiting?

**Comment by aliabadi** [comment]
I did find a workaround of sorts.   I load the standard SDK through my podfile and when I open the new workspace I run the "Convert to Swift 3 Syntax" wizard and it works after that.  

Just have to build the project afterwards and everything is fine.

Thanks!

**Comment by idelfonsog2** [comment]
@aliabadi when you said the standard... Do you mean the Obj-C or the swift sdk?

**Pull request merged**